packages/teams-js/src/public/video.ts

```diff
@@ -216,4 +130,361 @@ export namespace video {
+     */
+    export interface PlaneLayout {
+      offset: number;
+      stride: number;
+    }
+
+    /**
+     * Align with the W3C spec: https://www.w3.org/TR/webcodecs/
+     */
+    export interface VideoFrameCopyToOptions {
+      layout?: PlaneLayout[] | undefined;
+      rect?: DOMRectInit | undefined;
+    }
+
+    /**
+     * Align with the W3C spec: https://www.w3.org/TR/webcodecs/
+     */
+    export interface VideoFrameInit {
+      alpha?: AlphaOption | undefined;
+      displayHeight?: number | undefined;
+      displayWidth?: number | undefined;
+      duration?: number | undefined;
+      timestamp?: number | undefined;
+      visibleRect?: DOMRectInit | undefined;
+    }
+
+    /**
+     * Align with the W3C spec: https://www.w3.org/TR/webcodecs/
+     */
+    export interface VideoFrameBufferInit {
+      codedHeight: number;
+      codedWidth: number;
+      colorSpace?: VideoColorSpaceInit | undefined;
+      displayHeight?: number | undefined;
+      displayWidth?: number | undefined;
+      duration?: number | undefined;
+      format: VideoPixelFormat;
+      layout?: PlaneLayout[] | undefined;
+      timestamp: number;
+      visibleRect?: DOMRectInit | undefined;
+    }
+
+    /**
+     * VideoFrame definition, align with the W3C spec: https://www.w3.org/TR/webcodecs/
+     */
+    export interface VideoFrame {
+      readonly codedHeight: number;
+      readonly codedRect: DOMRectReadOnly | null;
+      readonly codedWidth: number;
+      readonly colorSpace: VideoColorSpace;
+      readonly displayHeight: number;
+      readonly displayWidth: number;
+      readonly duration: number | null;
+      readonly format: VideoPixelFormat | null;
+      readonly timestamp: number | null;
+      readonly visibleRect: DOMRectReadOnly | null;
+      allocationSize(options?: VideoFrameCopyToOptions): number;
+      clone(): VideoFrame;
+      close(): void;
+      copyTo(destination: AllowSharedBufferSource, options?: VideoFrameCopyToOptions): Promise<PlaneLayout[]>;
+    }
+
+    /**
+     * VideoFrame definition, align with the W3C spec: https://www.w3.org/TR/webcodecs/
+     */
+    // eslint-disable-next-line strict-null-checks/all
+    declare const VideoFrame: {
+      prototype: VideoFrame;
+      new (source: CanvasImageSource, init?: VideoFrameInit): VideoFrame;
+      new (data: AllowSharedBufferSource, init: VideoFrameBufferInit): VideoFrame;
+    };
+
+    /**
+     * @beta
+     * Checks if video.mediaStream capability is supported by the host
+     * @returns boolean to represent whether the video.medisStream capability is supported
+     *
+     * @throws Error if {@linkcode app.initialize} has not successfully completed
+     *
+     */
+    export function isSupported(): boolean {
+      return ensureInitialized(runtime) && isTextureStreamAvailable() && !!runtime.supports.video?.mediaStream;
+    }
+
+    function isTextureStreamAvailable(): boolean {
+      return (
+        typeof window !== 'undefined' &&
```

Comment:
Can you please use the inServerSideRenderingEnvironment function to test for the existence of window (either here or in other places in the change), to make the intent clearer? The function is currently in app.ts but you can move it into a separate file so it can be more generally used. E.g. create a new file in src/internal/serverSideRendering.ts and put that one function into it.

Reply:
Done